services/web/server/src/simcore_service_webserver/clusters/handlers.py

```diff
@@ -0,0 +1,144 @@
+import logging
+from typing import List
+
+from aiohttp import web
+from models_library.users import GroupID, UserID
+from servicelib.rest_utils import extract_and_validate
+
+from .._meta import api_version_prefix
+from ..groups_api import list_user_groups
+from ..login.decorators import RQT_USERID_KEY, login_required
+from ..security_decorators import permission_required
+from .db import ClustersRepository
+from .exceptions import ClusterAccessForbidden, ClusterNotFoundError
+from .models import Cluster, ClusterCreate, ClusterPatch
+
+logger = logging.getLogger(__name__)
+
+# API ROUTES HANDLERS ---------------------------------------------------------
+routes = web.RouteTableDef()
+
+
+@routes.get(f"/{api_version_prefix}/clusters", name="list_clusters_handler")
+@login_required
+@permission_required("clusters.read")
+async def list_clusters_handler(request: web.Request) -> web.Response:
+    await extract_and_validate(request)
+    user_id: UserID = request[RQT_USERID_KEY]
+
+    primary_group, std_groups, all_group = await list_user_groups(request.app, user_id)
+
+    clusters_repo = ClustersRepository(request)
+
+    clusters_list: List[Cluster] = await clusters_repo.list_clusters(
+        GroupID(primary_group["gid"]),
+        [GroupID(g["gid"]) for g in std_groups],
+        GroupID(all_group["gid"]),
+    )
+
+    data = [d.dict(by_alias=True) for d in clusters_list]
+
+    return web.json_response(data={"data": data})
+
+
+@routes.post(f"/{api_version_prefix}/clusters", name="create_cluster_handler")
+@login_required
+@permission_required("clusters.create")
+async def create_cluster_handler(request: web.Request) -> web.Response:
```

Comment:
Why no error handling for creating a cluster? If ClusterCreate fails, as a user I'd prefer a nice error message.

Reply:
what do you mean a nice error message??? not sure we have that anywhere. and is there not supposed to be a middleware for unexpected errors already?

Reply:
Yes there is a middleware to handle unexpected errors. But in this case failing to create a cluster because one of the parameters is wrong (say it already exists). As a user, I would like some of this basic cases to be handled more gracefully.

Reply:
what do you mean it already exists? I am not sure I understand. How can it already exist? the ID is created by the database.

Reply:
Had a better look at it. Possible errors will arise with db connection. I guess it's fine, they are handled like this elsewhere.